app/interactors/weetInteractor.js

```diff
@@ -0,0 +1,20 @@
+const weetService = require('../services/weetsService');
+const userService = require('../services/userService');
+const errorMessages = require('../constants/errorMessages');
+const { validationError, databaseError } = require('../errors');
+const logger = require('../logger');
+
+exports.createWeet = async body => {
+  const user = await userService.findUserByEmail(body.email);
+  if (!user) throw databaseError(errorMessages.userNotFound);
```

Comment:
This should not be a database error. Instead of that, you could create a new custom error like `notFoundError` and map it to 404 HTTP code